de.metas.purchasecandidate.base/src/main/java/de/metas/purchasecandidate/BPPurchaseScheduleRepository.java

```diff
@@ -246,4 +270,96 @@ private AllBPPurchaseSchedule(@NonNull final BPartnerId bpartnerId, @NonNull fin
 					.findFirst();
 		}
 	}
+
+	public void changeLeadTimeOffset(final BPartnerId bpartnerId, int leadTimeOffset)
+	{
+		final BPPurchaseSchedule bpPurchaseSchedule = getByBPartnerIdAndValidFrom(bpartnerId, SystemTime.asLocalDate())
+														.orElse(null);
```

Comment:
use `orElseThrow`

Reply:
also, now I saw where are u using it.
I would suggest to have it something like 
public BPPurchaseSchedule  changeLeadTimeOffset(final BPPurchaseSchedule bpPurchaseSchedule, Duration leadTimeOffset)

or 

public BPPurchaseSchedule  changeLeadTimeOffset(final BPPurchaseScheduleId bpPurchaseScheduleId, Duration leadTimeOffset)


Your approach is not performant (u are loading the public BPPurchaseSchedule) again, nor precise (might be that u would get other purchase schedule.

More, please avoid SystemTime.asLocalDate() in your APIs. It's OK to use it in some "client" code but NOT in APIs. And even more, it was not obvious at all.
If would be better to have it as parameter.